contracts/external/proxies/CETHProxy.sol

```diff
@@ -0,0 +1,90 @@
+/**
+ * SPDX-License-Identifier: UNLICENSED
+ */
+pragma solidity =0.6.10;
+pragma experimental ABIEncoderV2;
+
+import {ReentrancyGuard} from "../../packages/oz/ReentrancyGuard.sol";
+import {SafeERC20} from "../../packages/oz/SafeERC20.sol";
+import {ERC20Interface} from "../../interfaces/ERC20Interface.sol";
+import {Actions} from "../../libs/Actions.sol";
+import {Controller} from "../../Controller.sol";
+import {CETHInterface} from "../../interfaces/CETHInterface.sol";
+
+/**
+ * @title CETHProxy
+ * @author Opyn Team
+ * @dev Contract for wrapping cToken before minting options
+ */
+contract CETHProxy is ReentrancyGuard {
+    using SafeERC20 for ERC20Interface;
+    using SafeERC20 for CETHInterface;
+
+    Controller public controller;
+    address public marginPool;
+    address public cethAddress;
+
+    constructor(
+        address _controller,
+        address _marginPool,
+        address _cethAddress
+    ) public {
+        controller = Controller(_controller);
+        marginPool = _marginPool;
+        cethAddress = _cethAddress;
+    }
+
+    /**
+     * @notice execute a number of actions after minting some cTokens
+     * @dev a wrapper for the Controller operate function, to wrap uderlying to cToken before the excution.
+     * @param _actions array of actions arguments
+     * @param _underlying underlying asset
+     * @param _cToken the cToken to mint
+     */
+    function operate(
+        Actions.ActionArgs[] memory _actions,
+        address _underlying,
+        address _cToken
+    ) external payable nonReentrant {
+        require(_underlying == address(0), "CETHWrapUnwrap: ETH address other than address(0) specified");
+        require(_cToken == cethAddress, "CETHWrapUnwrap: Wrong cToken address specified for ETH");
```

Comment:
```suggestion
    function operate(
        Actions.ActionArgs[] memory _actions,
        address _underlying
    ) external payable nonReentrant {
        require(_underlying == address(0), "CETHWrapUnwrap: ETH address other than address(0) specified");
```
In case of cETH, I don't think we need to pass the ctoken type again, can just use predefined cethAddress